src/python/pants/build_graph/address.py

```diff
@@ -385,14 +389,12 @@ def sanitize(s: str) -> str:
         return f"{prefix}{path}{target}{params}{generated}"
 
     def maybe_convert_to_target_generator(self) -> Address:
```

Comment:
I'm having a hard time reasoning about this with parametrization. If it was parametrized, then there is no base target, right?

Also this is a misnomer parametrization - that's not a "target generator".

Reply:
> If it was parametrized, then there is no base target, right?

There is still a base target `Address`, which is used as the key for `_TargetParametrizations`, which is what generates generated or parametrized targets.

I'll move this change into #14408.